src/deque.jl

```diff
@@ -322,6 +323,7 @@ function Base.popfirst!(d::Deque{T}) where T
     @assert head.back >= head.front
 
     @inbounds x = head.data[head.front]
+    Base._unsetindex!(head.data, head.front)
```

Comment:
i wonder if these lines are worthy of a comment? maybe even just 
```suggestion
    Base._unsetindex!(head.data, head.front) # see issue/884
```